lib/opensusebasetest.pm

```diff
@@ -1006,9 +1006,12 @@ sub select_serial_terminal {
         }
     } elsif ($backend =~ /^(ikvm|ipmi|spvm)$/) {
         $console = 'root-ssh';
+    } elsif ($backend =~ /^(generalhw)$/ && !defined(get_var('GENERAL_HW_VNC_IP'))) {
```

Comment:
why the "!defined" part? If it isn't for this part, can you combine it with the above and use a function from lib/Utils/Backends.pm , e.g. "has_ttys"?

Reply:
This would probably be better, indeed.

Reply:
ok, can you add a method that is suitable then?

Reply:
New function: `is_generalhw_no_vnc` ? Not sure which package would fit the best to add it, any suggestion?

Reply:
No, I mean a function that describes why we want to have a root-ssh but is not "has_ttys". What is the commonality between these and why does `has_ttys` not fit here?

Reply:
It would be `has_serial_over_ssh`.

Reply:
sure, that's a good idea. Any name is better than a "if(foo|bar|baz)" not knowing why certain backends are specified there and others not :)

Reply:
Updated.